pkg/terra/median_contract.go

```diff
@@ -79,36 +91,33 @@ func (ct *Contract) LatestTransmissionDetails(
 }
 
 // LatestRoundRequested fetches the latest round requested by filtering event logs
-func (ct *Contract) LatestRoundRequested(ctx context.Context, lookback time.Duration) (
+func (ct *MedianContract) LatestRoundRequested(ctx context.Context, lookback time.Duration) (
 	configDigest types.ConfigDigest,
 	epoch uint32,
 	round uint8,
 	err error,
 ) {
 	// calculate start block
-	latestBlock, blkErr := ct.terra.clientCtx.Client.Block(ctx, nil)
+	latestBlock, blkErr := ct.r.LatestBlock()
 	if blkErr != nil {
 		err = blkErr
 		return
 	}
-	blockNum := uint64(latestBlock.Block.Height) - uint64(lookback.Seconds())/BlockRate
-	queryStr := fmt.Sprintf("tx.height > %d AND wasm-new_round.contract_address='%s'", blockNum, ct.ContractAddress)
-	res, err := ct.terra.clientCtx.Client.TxSearch(ctx, queryStr, false, nil, nil, "desc")
+	blockNum := uint64(latestBlock.Block.Header.Height) - uint64(lookback.Seconds())/BlockRate
+	res, err := ct.r.TxsEvents([]string{fmt.Sprintf("tx.height>=%d", blockNum+1), fmt.Sprintf("wasm-new_round.contract_address='%s'", ct.contract.String())})
 	if err != nil {
 		return
 	}
-	if len(res.Txs) == 0 || res.TotalCount == 0 {
+	if len(res.TxResponses) == 0 {
 		return
 	}
-
-	// use the last one, should be the latest tx with event
```

Comment:
this is actually a bug fix - in one place it was assuming that txes were coming back in descending order and here it was assuming ascending order. Now both TxEvents searches use descending